packages/block-editor/src/components/block-tools/use-show-block-tools.js

```diff
@@ -34,38 +34,37 @@ export function useShowBlockTools() {
 				getSelectedBlockClientId() ||
 				getFirstMultiSelectedBlockClientId();
 
-			const { name = '', attributes = {} } = getBlock( clientId ) || {};
 			const editorMode = __unstableGetEditorMode();
-			const hasSelectedBlock = clientId && name;
-			const isEmptyDefaultBlock = isUnmodifiedDefaultBlock( {
-				name,
-				attributes,
-			} );
-			const _showEmptyBlockSideInserter =
+			const block = getBlock( clientId );
+			const hasSelectedBlock = clientId && !! block?.name;
+			const isEmptyDefaultBlock =
+				hasSelectedBlock && isUnmodifiedDefaultBlock( block );
```

Comment:
The `isUnmodifiedDefaultBlock` expects the block object. Even though it currently only uses `attribute` and `name` properties, I think this is more future proof.

Reply:
Follow up at https://github.com/WordPress/gutenberg/pull/59450